### ISSUE TITLE
MySQL keyword 'collate' not support in parser

### ISSUE THREAD

**Issue by LepX** [vitessio/vitess#3757]
```
ERROR: logging before flag.Parse: W0319 18:50:07.767923   42007 ast.go:55] ignoring error parsing DDL 'CREATE TABLE `test` (
  `id` int(64) NOT NULL AUTO_INCREMENT,
  `meta_id` int(64) NOT NULL,
  `data_meta` int(64) NOT NULL,
  `start_time` timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
  `end_time` timestamp NOT NULL DEFAULT '0000-00-00 00:00:00',
  `host_ip` char(32) COLLATE utf8_unicode_ci NOT NULL,
  `result` enum('0','1','2') COLLATE utf8_unicode_ci DEFAULT '2',
  PRIMARY KEY (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci COMMENT='test'': syntax error at position 347 near 'collate'
```

**Comment by sougou** [comment]
@demmer looks like the ignore mode didn't get triggered for this construct?

**Comment by LepX** [comment]
@sougou 
so thanks for your response. 
sqlparser will give a great help to my project. 
I took the test with some SQLs and I found this error message, and i could not  get the expr for this SQL.
Is there a supportive plan for this keyword?
or if not, is there any way to ignore 'COLLATE' and get the tree? 
or i just remove the keyword in some other way.

best wishes.

**Comment by demmer** [comment]
@LepX Oops -- sorry about that. It turns out this is an issue only with enum fields. It should be fixed in #3763 

@sougou the ignore mode was triggered as you can see in the error message:
> ERROR: logging before flag.Parse: W0319 18:50:07.767923   42007 ast.go:55] ignoring error parsing DDL 'CREATE TABLE `test` (

**Comment by LepX** [comment]
tks a lot 👍
